backend/src/graphql/server.ts

```diff
@@ -0,0 +1,56 @@
+import { ApolloServer } from '@apollo/server';
+import { resolvers } from './resolvers';
+import {
+  constraintDirective,
+  constraintDirectiveTypeDefs
+} from 'graphql-constraint-directive';
+import DataLoader from 'dataloader';
+import { authorService } from 'src/services/author';
+import { bookService } from 'src/services/book';
+import { makeExecutableSchema } from '@graphql-tools/schema';
+import typeDefs from './schema.graphql';
+
+let schema = makeExecutableSchema({
+  typeDefs: [constraintDirectiveTypeDefs, typeDefs],
+  resolvers
+});
+schema = constraintDirective()(schema);
+
+const gqlServer = new ApolloServer({
+  schema
+});
+
+const getDataLoaders = () => {
+  const authorLoader = new DataLoader(async (ids: readonly string[]) => {
+    authorLoader.clearAll(); // clear out cache for fetching the most updated data from the source
+    const authorMap = {};
+    for (const id of ids) {
+      const author = await authorService.getById(id);
+      if (author) {
+        authorMap[id] = author;
+      } else {
+        authorMap[id] = null;
+      }
+    }
```

Comment:
In a Dataloader, what we want to do is convert several DB calls into one DB call. The current implementation makes one DB call per id. We should change the implementation so that we are making only a single DB call for fetching all the authors.